Remove canon_host-related checks from CI

CPerezz opened the pull request:
Resolves: #136

Commit:
Remove canon_host-related checks from CI

Resolves: #136

jules reviewed (approved):
LGTM, merge when the CI passes!

Commit:
Update rustfmt toolchain version

Commit:
Remove anyhow leftovers from docs

Commit:
Use alloc feature by default for plonk & hades

Commit:
Change the feature requirements of the crate

The requirements were not in-line with the dependencies that we
are using and it's feature sets.

Therefore, we no longer need to restrict the zk part of things
to `std` and instead can always have it enabled via using `alloc`
by default.

Resolves: #138

Commit:
Fix header indentation for dusk-analyzer

ZER0 reviewed (approved):
LGTM, but is there is a specific reason why you update the rustfmt version? Consider that if we change the rustfmt vesion in one repo, we have to change in all repos.

Commit:
Remove unnecessary cfg std flag

Commit:
Update CHANGELOG.md

CPerezz committed:
Merge pull request #139 from dusk-network/features-fix

Change the feature requirements of the crate

Commit:
Rollback fmt rustc version to nightly-2020-10-25
